src/query/string_opfunc.c

```diff
@@ -3955,14 +3956,15 @@ db_json_search_dbval (DB_VALUE * result, DB_VALUE * args[], const int num_args)
       return error_code;
     }
 
-  char *find_all_str = db_get_string (args[1]);
-  bool find_all = false;
+  std::string find_all_str (db_get_string (args[1]));
```

Comment:
I think db_json_contains_path also needs same checks. Maybe it is better to make a function of all/one checks and call it in both places.